### PR TITLE
[Dashboard]: add send telemetry to admin settings

### LINKED ISSUE #7573: [Dashboard]: create telemetry screen in admin panel
Further to #7572, the same opt-out question should appear in an admin panel to allow for admin users to change their status after initial installation

Once complete, the database should be change so that the `sendTelemetry` defaults to `true` #7591

### PR DESCRIPTION
**Pull request by MrSimonEmms**
## Description

This adds the "send telemetry" checkbox to the Admin dashboard and sets the default option to `true` - the front-end for #7591

It also moves the `guardAdminAccess` method from the EE `gitpod-server-impl.ts` to the one in `src`. This is because we're going to open up more dashboard administrative options to users regardless of whether they have an enterprise license or not - clearly, the option to disable telemetry should be available to all users.

## Related Issue(s)

Fixes #7573

## How to test

Go to the [Dashboard](https://sje-installation-admin-dashboard.staging.gitpod-dev.com/workspaces) -> Settings and click the "send telemetry" checkbox

## Release Notes

```release-note
[Dashboard]: add send telemetry to admin settings
```

### CODE REVIEW

**Comment by roboquat** [comment]
@MrSimonEmms: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **mrsimonemms** and additionally assign **jldec** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jldec` in a comment when ready.

Associated issue: *#7573*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/installation-telemetry/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/installation-telemetry/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**
- **[installer/pkg/components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **mrsimonemms** and additionally assign **jldec** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jldec` in a comment when ready.

Associated issue: *#7591*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/installation-telemetry/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/installation-telemetry/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**
- **[installer/pkg/components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jldec** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jldec` in a comment when ready.

Associated issue: *#7591*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jldec** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jldec` in a comment when ready.

Associated issue: *#7573*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/admin/Settings.tsx
Comment (geropl) [inline components/dashboard/src/admin/Settings.tsx:26]: 1st questions that pops up here is: "What exactly is sent?" Either explained here, or with a ref to docs.

Comment (MrSimonEmms) [reply components/dashboard/src/admin/Settings.tsx:26]: Good question. We had a @gitpod-io/engineering-self-hosted [discussion on this](https://gitpod.slack.com/archives/C01KLC56NP7/p1642696983023300) and agreed that wording was enough to be going on with.

Is there a way of putting a link to the [privacy policy](https://www.gitpod.io/privacy) in the description?

File: components/gitpod-protocol/src/admin-protocol.ts
Comment (geropl) [inline components/gitpod-protocol/src/admin-protocol.ts:33]: Instead of having a method here for every boolean flag, we should implement something akin [`updateLoggedInUser`](https://github.com/gitpod-io/gitpod/blob/0951ced5ebfb2e1a822b7c3eaa4855f7cad7724c/components/server/src/workspace/gitpod-server-impl.ts#L283-L298).

This avoids the necessity to update the API surface and implementation on every change.

File: components/gitpod-protocol/src/gitpod-service.ts
Comment (geropl) [inline components/gitpod-protocol/src/gitpod-service.ts:135]: See [comment above](https://github.com/gitpod-io/gitpod/pull/7732/files#r791663353)

File: components/gitpod-db/src/installation-admin-db.ts
Comment (geropl) [inline components/gitpod-db/src/installation-admin-db.ts:12]: See [comment below](https://github.com/gitpod-io/gitpod/pull/7732/files#r791663353).

File: components/server/src/workspace/gitpod-server-impl.ts
Comment (geropl) [inline components/server/src/workspace/gitpod-server-impl.ts:568]: :+1:

File: components/gitpod-protocol/src/admin-protocol.ts
Comment (geropl) [reply components/gitpod-protocol/src/admin-protocol.ts:33]: Note that it's important to have an explicit allowlist for fields to update, otherwise this is an attack vector.

One way to solve this without further duplication would be to change `installation-admin-protocol.ts` to sth like this:
```
/**
 * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
 * Licensed under the GNU Affero General Public License (AGPL).
 * See License-AGPL.txt in the project root for license information.
 */


const InstallationAdminSettingsPrototype = {
    sendTelemetry: true,
};
export type InstallationAdminSettings = typeof InstallationAdminSettingsPrototype;
export namespace InstallationAdminSettings {
    export function fields(): (keyof InstallationAdminSettings)[] {
        return Object.keys(InstallationAdminSettingsPrototype) as (keyof InstallationAdminSettings)[];
    }
}

export interface InstallationAdmin {
    id: string;
    settings: InstallationAdminSettings;
}
export namespace InstallationAdmin {
    export function createDefault(): InstallationAdmin {
        return {
            id: "",
            settings: {
                ...InstallationAdminSettingsPrototype,
            }
        };
    }
}
```
And use the `InstallationAdminSettings.fields()` method to iterate the shape's properties like in `updateLggedInUser` with the explicit list.

**Comment by gtsiolis** [comment]
Looking at UX changes in this now! 👀 

Thanks for the ping, @geropl. 🏓

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** after the PR has been reviewed.

Associated issue: *#7591*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/admin/Settings.tsx
Comment (gtsiolis) [inline components/dashboard/src/admin/Settings.tsx:28]: **suggestion**: If we change the title as described above we could go with one of the following proposals depending if we have any relevant docs in place:

Option :a:
```suggestion
                    desc={<span>To help improve Gitpod, service ping will periodically collect anonymous usage information. <a className="gp-link" href="https://www.gitpod.io/docs/LINK-TO-DOCS-OR-PRIVACY-POLICY">Learn more</a></span>}
```

Option :b:
```suggestion
                    desc={<span>To help improve Gitpod, service ping will periodically collect anonymous usage information. Read our <a className="gp-link" href="https://www.gitpod.io/privacy">Read our Privavy Policy</a></span>}
```

File: components/dashboard/src/App.tsx
Comment (gtsiolis) [inline components/dashboard/src/App.tsx:56]: **issue(non-blocking)**: Just wanted to add a comment that I could not access the admin panel evern after trying to add a user as an admin using this documentation[[1](https://www.notion.so/gitpod/How-we-develop-e8ce7e562478458a9223eb9b797415b2#d0f67538075f4050ba7c2551c7160f3b)].

**Review by gtsiolis — APPROVED** [review]
Thanks, @MrSimonEmms! 🔮 

Left some minor comments. ✔️ 

Approving to unblock merging but holding in case we'd like to address any of the UX feedback added below. 🏁 

/hold

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 863e9a730d4ac464a78bb6ae6afee173f407b80c</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7732#pullrequestreview-862274333" title="Approved">gtsiolis</a>*

Associated issue: *#7591*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [gtsiolis]
- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [gtsiolis]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [gtsiolis]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [gtsiolis]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7732#pullrequestreview-862274333" title="Approved">gtsiolis</a>*

Associated issue: *#7591*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [gtsiolis]
- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [gtsiolis]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [gtsiolis]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [gtsiolis]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by MrSimonEmms** [comment]
@geropl I've updated as requested. I believe that's as requested, so please let me know if I've misunderstood any of your change requests.

@gtsiolis I've added in the link - thank you. I've slightly changed the text so it's more in keeping with what @gitpod-io/engineering-self-hosted agreed on.

image

File: components/dashboard/src/App.tsx
Comment (MrSimonEmms) [reply components/dashboard/src/App.tsx:56]: I'm not sure about adding a user as admin. In my environment where my user is an admin by default, it shows ok. As you say, I think it's probably ok as there will always be the initial admin. We can always raise as a bugfix if a wider problem

File: components/dashboard/src/admin/Settings.tsx
Comment (MrSimonEmms) [reply components/dashboard/src/admin/Settings.tsx:27]: Yes we can. At this stage, we only send a randomly generated ID and the version tag - see the [code](https://github.com/gitpod-io/gitpod/blob/main/components/installation-telemetry/cmd/send.go#L53)

File: components/dashboard/src/App.tsx
Comment (gtsiolis) [reply components/dashboard/src/App.tsx:56]: Yes, the first user always becomes an admin. That was more like an observation that I could not test this in the preview environment properly either because the script to make someone admin does not work or because we could improve our code review process.

File: components/dashboard/src/admin/Settings.tsx
Comment (gtsiolis) [reply components/dashboard/src/admin/Settings.tsx:27]: Nice! Thanks for confirming, @MrSimonEmms!

Comment (MrSimonEmms) [reply components/dashboard/src/admin/Settings.tsx:27]: #7820

**Comment by gtsiolis** [comment]
@MrSimonEmms could you make me an admin in the preview env so that I can see the UX changes in action?

**Comment by MrSimonEmms** [comment]
@gtsiolis done. I've added you as well Gero

File: components/dashboard/src/App.tsx
Comment (MrSimonEmms) [reply components/dashboard/src/App.tsx:56]: I've raised PR #7840 with a fix for leeway

File: components/dashboard/src/admin/Settings.tsx
Comment (gtsiolis) [inline components/dashboard/src/admin/Settings.tsx:26]: **suggestion**: What do you think of adding a section header for this setting as we do for all other settings across the product? Not sure if the code suggestion here would work as expected. 😇 

```suggestion
                <h3>Usage Statistics</h3>
                <CheckBox
```

File: components/dashboard/src/admin/Settings.tsx
Comment (gtsiolis) [inline components/dashboard/src/admin/Settings.tsx:28]: **suggestion(non-blocking)**: To re-use the pattern of the rest of help links (_Learn more_ links) across the product we could hyperlink the whole action here. What do you think? 

```suggestion
                    desc={<span>This is used to provide insights on how you use your cluster so we can provide a better overall experience. <a className="gp-link" href="https://www.gitpod.io/privacy">Read our privacy policy</a></span>}
```

**Review by gtsiolis — APPROVED** [review]
UX works like a charm, @MrSimonEmms! 

Thanks for making me admin to see this in actions and opening the follow up PR in https://github.com/gitpod-io/gitpod/pull/7840. 🏀 

Left two minor suggestions but feel free to ignore these if needed.

| BEFORE | AFTER |
|-|-|
| Screenshot 2022-01-26 at 2 28 54 PM (2) | Screenshot 2022-01-26 at 2 28 48 PM (2) |

Approving to unblock merging but holding in case we'd like to address any of the pending comments. Your call. 🏓 

/hold

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: e094102bbfceed842f01750fef0ae4d1ecfb1e0a</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7732#pullrequestreview-863474737" title="Approved">gtsiolis</a>*

Associated issue: *#7591*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [gtsiolis]
- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [gtsiolis]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [gtsiolis]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [gtsiolis]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/server/src/workspace/gitpod-server-impl.ts
Comment (geropl) [inline components/server/src/workspace/gitpod-server-impl.ts:2033]: This should read `adminUpdateSettings` I guess. :+1:

File: components/server/src/workspace/gitpod-server-impl.ts
Comment (geropl) [inline components/server/src/workspace/gitpod-server-impl.ts:2020]: :heart:

File: components/server/src/workspace/gitpod-server-impl.ts
Comment (MrSimonEmms) [reply components/server/src/workspace/gitpod-server-impl.ts:2033]: Well spotted - will change

File: components/gitpod-db/src/typeorm/installation-admin-db-impl.ts
Comment (geropl) [inline components/gitpod-db/src/typeorm/installation-admin-db-impl.ts:56]: @MrSimonEmms This `...` might be an attack vector, depending on the implementation of `repo.save(...)`. (we had one with `.update` last fall).

I suggest we:
 - keep the spread operator (`...`) here
 - add an explicit loop over all fields in `api.adminUpdateSettings` (cmp. comment below)

File: components/gitpod-db/src/typeorm/installation-admin-db-impl.ts
Comment (geropl) [inline components/gitpod-db/src/typeorm/installation-admin-db-impl.ts:60]: `await` missing.

File: components/server/src/workspace/gitpod-server-impl.ts
Comment (geropl) [inline components/server/src/workspace/gitpod-server-impl.ts:2035]: Instead of calling this directly, we need to check the properties we expect in `settings` explicitly:
```
        const newSettings: Partial<Settings> = {};
        for (const p of InstallationAdminSettings.fields()) {
            if (p in settings) {
                (newSettings[p] as any) = settings[p];
            }
        }
       await this.installationAdminDb.setSettings(newSettings);
```

File: components/dashboard/src/admin/Settings.tsx
Comment (MrSimonEmms) [reply components/dashboard/src/admin/Settings.tsx:28]: You're the boss when it comes to the front-end - I'm happy to oblige with this

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

File: components/server/src/installation-admin/installation-admin-controller.ts
Comment (geropl) [inline components/server/src/installation-admin/installation-admin-controller.ts:21]: nit: `res.status(200).json(data);` also sets the proper content header. Feel free to ignore, as you're managing the only client here :+1:

File: components/gitpod-db/src/installation-admin-db.ts
Comment (geropl) [inline components/gitpod-db/src/installation-admin-db.ts:12]: nit: this requires only `settings: Partial<InstallationAdmingSettings>`

**Review by geropl — APPROVED** [review]
Nice @MrSimonEmms :rocket: 

I left two more nits, but feel free to ignore to proceed here!

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: bbd9a19ca4bec142f1ed26d44a299b1131e68f82</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7732#pullrequestreview-863630053" title="Approved">geropl</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7732#pullrequestreview-863474737" title="Approved">gtsiolis</a>*

Associated issue: *#7591*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [geropl,gtsiolis]
- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [geropl,gtsiolis]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [geropl,gtsiolis]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl,gtsiolis]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by MrSimonEmms** [comment]
@geropl nits fixed

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 6271ab208f8968504a76720a3a148d9360d078f9</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7732#issuecomment-1022271166" title="LGTM">geropl</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7732#pullrequestreview-863474737" title="Approved">gtsiolis</a>*

Associated issue: *#7591*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [geropl,gtsiolis]
- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [geropl,gtsiolis]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [geropl,gtsiolis]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl,gtsiolis]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by MrSimonEmms** [comment]
@geropl @gtsiolis thank you both for your help/patience on these PRs 🙏🏻  🎉

**Pull request merged**
